[build] QuaZip fails to build with /W4 on MSVC

...or `shadow` and `missing-declarations` on Linux.

This is a pretty significant problem for projects that enforce W4 or Werror.

```
D:\a\PR\PR\eden\build\_deps\quazip-src\quazip\unzip.c(1057): error C4456: declaration of 'uL' hides previous local declaration
D:\a\PR\PR\eden\build\_deps\quazip-src\quazip\unzip.c(912): note: see declaration of 'uL'
D:\a\PR\PR\eden\build\_deps\quazip-src\quazip\unzip.c(1766): warning C4244: '=': conversion from 'int' to 'char', possible loss of data
```

It's possible to ignore it on GCC/clang, but I have frankly no idea how to do so on MSVC. Ideally errors like this are fixed. I had a patch to fix it but lost it after some CMake shenanigans, I'll try to make a PR soon.

This is interesting, upon a CI build, these errors pop up;

```
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/minizip_crypt.h: In function ‘crypthead’:
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/minizip_crypt.h:93:12: warning: old-style function definition [-Wold-style-definition]
   93 | static int crypthead(passwd, buf, bufSize, pkeys, pcrc_32_tab, crcForCrypting)
      |            ^~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/zip.c: At top level:
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/zip.c:648:5: error: no previous declaration for ‘LoadCentralDirectoryRecord’ [-Werror=missing-declarations]
  648 | int LoadCentralDirectoryRecord(zip64_internal* pziinit)
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/zip.c:972:5: error: no previous declaration for ‘Write_LocalFileHeader’ [-Werror=missing-declarations]
  972 | int Write_LocalFileHeader(zip64_internal* zi, const char* filename,
      |     ^~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/zip.c:1821:5: error: no previous declaration for ‘Write_Zip64EndOfCentralDirectoryLocator’ [-Werror=missing-declarations]
 1821 | int Write_Zip64EndOfCentralDirectoryLocator(zip64_internal* zi, ZPOS64_T zip64eocd_pos_inzip)
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/zip.c:1843:5: error: no previous declaration for ‘Write_Zip64EndOfCentralDirectoryRecord’ [-Werror=missing-declarations]
 1843 | int Write_Zip64EndOfCentralDirectoryRecord(zip64_internal* zi, uLong size_centraldir, ZPOS64_T centraldir_pos_inzip)
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/zip.c:1882:5: error: no previous declaration for ‘Write_EndOfCentralDirectoryRecord’ [-Werror=missing-declarations]
 1882 | int Write_EndOfCentralDirectoryRecord(zip64_internal* zi, uLong size_centraldir, ZPOS64_T centraldir_pos_inzip)
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/zip.c:1930:5: error: no previous declaration for ‘Write_GlobalComment’ [-Werror=missing-declarations]
 1930 | int Write_GlobalComment(zip64_internal* zi, const char* global_comment)
      |     ^~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
[2736/2894] Building CXX object _deps/quazip-build/quazip/CMakeFiles/QuaZip.dir/JlCompress.cpp.o
[2737/2894] Building CXX object _deps/quazip-build/quazip/CMakeFiles/QuaZip.dir/qioapi.cpp.o
FAILED: _deps/quazip-build/quazip/CMakeFiles/QuaZip.dir/qioapi.cpp.o 
/usr/sbin/c++ -DARCHITECTURE_x86_64=1 -DHAVE_BZIP2 -DNDEBUG -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_NO_DEBUG -DQUAZIP_STATIC -DYUZU_ROOM -DYUZU_UNIX=1 -D_FILE_OFFSET_BITS=64 -I/__w/PR/PR/eden/build/_deps/quazip-build/quazip -I/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip -I/__w/PR/PR/eden/build/_deps/quazip-build/quazip/QuaZip_autogen/include -I/__w/PR/PR/eden/src/. -I/__w/PR/PR/eden/.cache/cpm/quazip/045f -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore5Compat -march=x86-64-v3 -O3 -O3 -DNDEBUG -std=gnu++17 -fwrapv -Werror=all -Werror=extra -Werror=missing-declarations -Werror=shadow -Werror=unused -Wno-attributes -Wno-invalid-offsetof -Wno-unused-parameter -Wno-missing-field-initializers -mcx16 -Wno-array-bounds -Wno-stringop-overread -Wno-stringop-overflow -mno-direct-extern-access -MD -MT _deps/quazip-build/quazip/CMakeFiles/QuaZip.dir/qioapi.cpp.o -MF _deps/quazip-build/quazip/CMakeFiles/QuaZip.dir/qioapi.cpp.o.d -o _deps/quazip-build/quazip/CMakeFiles/QuaZip.dir/qioapi.cpp.o -c /__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:99:18: error: no previous declaration for ‘void* qiodevice_open_file_func(voidpf, voidpf, int)’ [-Werror=missing-declarations]
   99 | voidpf ZCALLBACK qiodevice_open_file_func (
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:150:17: error: no previous declaration for ‘uLong qiodevice_read_file_func(voidpf, voidpf, void*, uLong)’ [-Werror=missing-declarations]
  150 | uLong ZCALLBACK qiodevice_read_file_func (
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:168:17: error: no previous declaration for ‘uLong qiodevice_write_file_func(voidpf, voidpf, const void*, uLong)’ [-Werror=missing-declarations]
  168 | uLong ZCALLBACK qiodevice_write_file_func (
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:185:17: error: no previous declaration for ‘uLong qiodevice_tell_file_func(voidpf, voidpf)’ [-Werror=missing-declarations]
  185 | uLong ZCALLBACK qiodevice_tell_file_func (
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:202:20: error: no previous declaration for ‘ZPOS64_T qiodevice64_tell_file_func(voidpf, voidpf)’ [-Werror=missing-declarations]
  202 | ZPOS64_T ZCALLBACK qiodevice64_tell_file_func (
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:217:15: error: no previous declaration for ‘int qiodevice_seek_file_func(voidpf, voidpf, uLong, int)’ [-Werror=missing-declarations]
  217 | int ZCALLBACK qiodevice_seek_file_func (
      |               ^~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:252:15: error: no previous declaration for ‘int qiodevice64_seek_file_func(voidpf, voidpf, ZPOS64_T, int)’ [-Werror=missing-declarations]
  252 | int ZCALLBACK qiodevice64_seek_file_func (
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:287:15: error: no previous declaration for ‘int qiodevice_close_file_func(voidpf, voidpf)’ [-Werror=missing-declarations]
  287 | int ZCALLBACK qiodevice_close_file_func (
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:297:15: error: no previous declaration for ‘int qiodevice_fakeclose_file_func(voidpf, voidpf)’ [-Werror=missing-declarations]
  297 | int ZCALLBACK qiodevice_fakeclose_file_func (
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/PR/PR/eden/.cache/cpm/quazip/045f/quazip/qioapi.cpp:306:15: error: no previous declaration for ‘int qiodevice_error_file_func(voidpf, voidpf)’ [-Werror=missing-declarations]
  306 | int ZCALLBACK qiodevice_error_file_func (
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
```

--- a/quazip/ioapi.h
+++ b/quazip/ioapi.h
@@ -181,6 +181,16 @@ typedef struct zlib_filefunc64_32_def_s
     seek_file_func      zseek32_file;
 } zlib_filefunc64_32_def;
 
+voidpf   ZCALLBACK qiodevice_open_file_func      OF((voidpf opaque, voidpf file, int mode));
+uLong    ZCALLBACK qiodevice_read_file_func      OF((voidpf opaque, voidpf stream, void* buf, uLong size));
+uLong    ZCALLBACK qiodevice_write_file_func     OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
+uLong    ZCALLBACK qiodevice_tell_file_func      OF((voidpf opaque, voidpf stream));
+ZPOS64_T ZCALLBACK qiodevice64_tell_file_func    OF((voidpf opaque, voidpf stream));
+int      ZCALLBACK qiodevice_seek_file_func      OF((voidpf opaque, voidpf stream, uLong offset, int origin));
+int      ZCALLBACK qiodevice64_seek_file_func    OF((voidpf opaque, voidpf stream, ZPOS64_T offset, int origin));
+int      ZCALLBACK qiodevice_close_file_func     OF((voidpf opaque, voidpf stream));
+int      ZCALLBACK qiodevice_fakeclose_file_func OF((voidpf opaque, voidpf stream));
+int      ZCALLBACK qiodevice_error_file_func     OF((voidpf opaque, voidpf stream));
 
 #define ZREAD64(filefunc,filestream,buf,size)     ((*((filefunc).zfile_func64.zread_file))   ((filefunc).zfile_func64.opaque,filestream,buf,size))
 #define ZWRITE64(filefunc,filestream,buf,size)    ((*((filefunc).zfile_func64.zwrite_file))  ((filefunc).zfile_func64.opaque,filestream,buf,size))

--- a/quazip/ioapi.h
+++ b/quazip/ioapi.h
@@ -126,9 +126,6 @@ extern "C" {
  #endif
 #endif
 
-
-
-
 typedef voidpf   (ZCALLBACK *open_file_func)      OF((voidpf opaque, voidpf file, int mode));
 typedef uLong    (ZCALLBACK *read_file_func)      OF((voidpf opaque, voidpf stream, void* buf, uLong size));
 typedef uLong    (ZCALLBACK *write_file_func)     OF((voidpf opaque, voidpf stream, const void* buf, uLong size));

--- a/quazip/minizip_crypt.h
+++ b/quazip/minizip_crypt.h
@@ -90,13 +90,14 @@ static void init_keys(const char* passwd,unsigned long* pkeys,const z_crc_t FAR 
 #    define ZCR_SEED2 3141592654UL     /* use PI as default pattern */
 #  endif
 
-static int crypthead(passwd, buf, bufSize, pkeys, pcrc_32_tab, crcForCrypting)
-    const char *passwd;         /* password string */
-    unsigned char *buf;         /* where to write header */
-    int bufSize;
-    unsigned long* pkeys;
-    const z_crc_t FAR * pcrc_32_tab;
-    unsigned long crcForCrypting;
+static int crypthead(
+    const char *passwd,         /* password string */
+    unsigned char *buf,         /* where to write header */
+    int bufSize,
+    unsigned long* pkeys,
+    const z_crc_t FAR * pcrc_32_tab,
+    unsigned long crcForCrypting
+)
 {
     int n;                       /* index in random header */
     int t;                       /* temporary */

--- a/quazip/quagzipfile.cpp
+++ b/quazip/quagzipfile.cpp
@@ -33,8 +33,8 @@ class QuaGzipFilePrivate {
     QString fileName;
     gzFile gzd;
     inline QuaGzipFilePrivate(): gzd(nullptr) {}
-    inline QuaGzipFilePrivate(const QString &fileName): 
-        fileName(fileName), gzd(nullptr) {}
+    inline QuaGzipFilePrivate(const QString &_fileName):
+        fileName(_fileName), gzd(nullptr) {}
     template<typename FileId> bool open(FileId id, 
         QIODevice::OpenMode mode, QString &error);
     gzFile open(int fd, const char *modeString);

--- a/quazip/quaziodevice.cpp
+++ b/quazip/quaziodevice.cpp
@@ -50,9 +50,9 @@ class QuaZIODevicePrivate {
     int doFlush(QString &error);
 };
 
-QuaZIODevicePrivate::QuaZIODevicePrivate(QIODevice *io, QuaZIODevice *q):
-  io(io),
-  q(q)
+QuaZIODevicePrivate::QuaZIODevicePrivate(QIODevice *_io, QuaZIODevice *_q):
+  io(_io),
+  q(_q)
 {
   zins.zalloc = (alloc_func) nullptr;
   zins.zfree = (free_func) nullptr;

--- a/quazip/quazip.cpp
+++ b/quazip/quazip.cpp
@@ -84,8 +84,8 @@ class QuaZipPrivate {
         return defaultFileNameCodec;
     }
     /// The constructor for the corresponding QuaZip constructor.
-    inline QuaZipPrivate(QuaZip *q):
-      q(q),
+    inline QuaZipPrivate(QuaZip *_q):
+      q(_q),
       fileNameCodec(getDefaultFileNameCodec()),
       commentCodec(QTextCodec::codecForLocale()),
       ioDevice(nullptr),
@@ -104,11 +104,11 @@ class QuaZipPrivate {
         lastMappedDirectoryEntry.pos_in_zip_directory = 0;
     }
     /// The constructor for the corresponding QuaZip constructor.
-    inline QuaZipPrivate(QuaZip *q, const QString &zipName):
-      q(q),
+    inline QuaZipPrivate(QuaZip *_q, const QString &_zipName):
+      q(_q),
       fileNameCodec(getDefaultFileNameCodec()),
       commentCodec(QTextCodec::codecForLocale()),
-      zipName(zipName),
+      zipName(_zipName),
       ioDevice(nullptr),
       mode(QuaZip::mdNotOpen),
       hasCurrentFile_f(false),
@@ -125,11 +125,11 @@ class QuaZipPrivate {
         lastMappedDirectoryEntry.pos_in_zip_directory = 0;
     }
     /// The constructor for the corresponding QuaZip constructor.
-    inline QuaZipPrivate(QuaZip *q, QIODevice *ioDevice):
-      q(q),
+    inline QuaZipPrivate(QuaZip *_q, QIODevice *_ioDevice):
+      q(_q),
       fileNameCodec(getDefaultFileNameCodec()),
       commentCodec(QTextCodec::codecForLocale()),
-      ioDevice(ioDevice),
+      ioDevice(_ioDevice),
       mode(QuaZip::mdNotOpen),
       hasCurrentFile_f(false),
       zipError(UNZ_OK),

--- a/quazip/quazipdir.cpp
+++ b/quazip/quazipdir.cpp
@@ -293,8 +293,8 @@ bool QuaZipDirComparator::operator()(const QuaZipFileInfo64 &info1,
 }
 
 template<typename TFileInfoList>
-bool QuaZipDirPrivate::entryInfoList(QStringList nameFilters,
-    QDir::Filters filter, QDir::SortFlags sort, TFileInfoList &result) const
+bool QuaZipDirPrivate::entryInfoList(QStringList _nameFilters,
+    QDir::Filters _filter, QDir::SortFlags sort, TFileInfoList &result) const
 {
     QString basePath = simplePath();
     if (!basePath.isEmpty())
@@ -305,12 +305,12 @@ bool QuaZipDirPrivate::entryInfoList(QStringList nameFilters,
     if (!zip->goToFirstFile()) {
         return zip->getZipError() == UNZ_OK;
     }
-    QDir::Filters fltr = filter;
+    QDir::Filters fltr = _filter;
     if (fltr == QDir::NoFilter)
         fltr = this->filter;
     if (fltr == QDir::NoFilter)
         fltr = QDir::AllEntries;
-    QStringList nmfltr = nameFilters;
+    QStringList nmfltr = _nameFilters;
     if (nmfltr.isEmpty())
         nmfltr = this->nameFilters;
     QSet<QString> dirsFound;

--- a/quazip/quazipdir.cpp
+++ b/quazip/quazipdir.cpp
@@ -32,8 +32,8 @@ see quazip/(un)zip.h files for details. Basically it's the zlib license.
 class QuaZipDirPrivate: public QSharedData {
     friend class QuaZipDir;
 private:
-    QuaZipDirPrivate(QuaZip *zip, const QString &dir = QString()):
-        zip(zip), dir(dir) {}
+    QuaZipDirPrivate(QuaZip *_zip, const QString &_dir = QString()):
+        zip(_zip), dir(_dir) {}
     QuaZip *zip;
     QString dir;
     QuaZip::CaseSensitivity caseSensitivity{QuaZip::csDefault};
@@ -199,8 +199,8 @@ static void QuaZipDir_convertInfoList(const QList<QuaZipFileInfo64> &from,
   */
 class QuaZipDirRestoreCurrent {
 public:
-    inline QuaZipDirRestoreCurrent(QuaZip *zip):
-        zip(zip), currentFile(zip->getCurrentFileName()) {}
+    inline QuaZipDirRestoreCurrent(QuaZip *_zip):
+        zip(_zip), currentFile(zip->getCurrentFileName()) {}
     inline ~QuaZipDirRestoreCurrent()
     {
         zip->setCurrentFile(currentFile);
@@ -219,7 +219,7 @@ class QuaZipDirComparator
         static QString getExtension(const QString &name);
         int compareStrings(const QString &string1, const QString &string2);
     public:
-        inline QuaZipDirComparator(QDir::SortFlags sort): sort(sort) {}
+        inline QuaZipDirComparator(QDir::SortFlags _sort): sort(_sort) {}
         bool operator()(const QuaZipFileInfo64 &info1, const QuaZipFileInfo64 &info2);
 };
 

--- a/quazip/quazipdir.h
+++ b/quazip/quazipdir.h
@@ -220,4 +220,9 @@ public:
     QDir::SortFlags sorting() const;
 };
 
+QuaZipFileInfo64 QuaZipDir_getFileInfo(QuaZip *zip,
+                                       bool *ok,
+                                       const QString &relativeName,
+                                       bool isReal);
+
 #endif // QUAZIP_QUAZIPDIR_H

--- a/quazip/quazipfile.cpp
+++ b/quazip/quazipfile.cpp
@@ -241,14 +241,14 @@ void QuaZipFile::setFileName(const QString& fileName, QuaZip::CaseSensitivity cs
   p->caseSensitivity=cs;
 }
 
-void QuaZipFilePrivate::setZipError(int zipError) const
+void QuaZipFilePrivate::setZipError(int _zipError) const
 {
   QuaZipFilePrivate *fakeThis = const_cast<QuaZipFilePrivate*>(this); // non-const
-  fakeThis->zipError=zipError;
-  if(zipError==UNZ_OK)
+  fakeThis->zipError = _zipError;
+  if(_zipError == UNZ_OK)
     q->setErrorString(QString());
   else
-    q->setErrorString(QuaZipFile::tr("ZIP/UNZIP API error %1").arg(zipError));
+    q->setErrorString(QuaZipFile::tr("ZIP/UNZIP API error %1").arg(_zipError));
 }
 
 bool QuaZipFile::open(OpenMode mode)

--- a/quazip/quazipfile.cpp
+++ b/quazip/quazipfile.cpp
@@ -80,8 +80,8 @@ class QuaZipFilePrivate {
       */
     void setZipError(int zipError) const;
     /// The constructor for the corresponding QuaZipFile constructor.
-    inline QuaZipFilePrivate(QuaZipFile *q):
-      q(q),
+    inline QuaZipFilePrivate(QuaZipFile *_q):
+      q(_q),
       zip(nullptr),
       caseSensitivity(QuaZip::csDefault),
       raw(false),
@@ -91,8 +91,8 @@ class QuaZipFilePrivate {
       internal(true),
       zipError(UNZ_OK) {}
     /// The constructor for the corresponding QuaZipFile constructor.
-    inline QuaZipFilePrivate(QuaZipFile *q, const QString &zipName):
-      q(q),
+    inline QuaZipFilePrivate(QuaZipFile *_q, const QString &zipName):
+      q(_q),
       caseSensitivity(QuaZip::csDefault),
       raw(false),
       writePos(0),
@@ -104,9 +104,9 @@ class QuaZipFilePrivate {
         zip=new QuaZip(zipName);
       }
     /// The constructor for the corresponding QuaZipFile constructor.
-    inline QuaZipFilePrivate(QuaZipFile *q, const QString &zipName, const QString &fileName,
+    inline QuaZipFilePrivate(QuaZipFile *_q, const QString &zipName, const QString &_fileName,
         QuaZip::CaseSensitivity cs):
-      q(q),
+      q(_q),
       raw(false),
       writePos(0),
       uncompressedSize(0),
@@ -115,15 +115,15 @@ class QuaZipFilePrivate {
       zipError(UNZ_OK)
       {
         zip=new QuaZip(zipName);
-        this->fileName=fileName;
+        this->fileName=_fileName;
         if (this->fileName.startsWith(QLatin1String("/")))
             this->fileName = this->fileName.mid(1);
         this->caseSensitivity=cs;
       }
     /// The constructor for the QuaZipFile constructor accepting a file name.
-    inline QuaZipFilePrivate(QuaZipFile *q, QuaZip *zip):
-      q(q),
-      zip(zip),
+    inline QuaZipFilePrivate(QuaZipFile *_q, QuaZip *_zip):
+      q(_q),
+      zip(_zip),
       raw(false),
       writePos(0),
       uncompressedSize(0),

--- a/quazip/quazipnewinfo.cpp
+++ b/quazip/quazipnewinfo.cpp
@@ -86,14 +86,14 @@ QuaZipNewInfo::QuaZipNewInfo(const QuaZipFileInfo64 &existing)
     QuaZipNewInfo_init(*this, existing);
 }
 
-QuaZipNewInfo::QuaZipNewInfo(const QString& name):
-  name(name), dateTime(QDateTime::currentDateTime()), internalAttr(0), externalAttr(0),
+QuaZipNewInfo::QuaZipNewInfo(const QString& _name):
+  name(_name), dateTime(QDateTime::currentDateTime()), internalAttr(0), externalAttr(0),
   uncompressedSize(0)
 {
 }
 
-QuaZipNewInfo::QuaZipNewInfo(const QString& name, const QString& file):
-  name(name), internalAttr(0), externalAttr(0), uncompressedSize(0)
+QuaZipNewInfo::QuaZipNewInfo(const QString& _name, const QString& file):
+  name(_name), internalAttr(0), externalAttr(0), uncompressedSize(0)
 {
   QFileInfo info(file);
   QDateTime lm = info.lastModified();
@@ -105,8 +105,8 @@ QuaZipNewInfo::QuaZipNewInfo(const QString& name, const QString& file):
   }
 }
 
-QuaZipNewInfo::QuaZipNewInfo(const QString& name, const QString& file, const QDateTime& dateTime):
-  name(name), dateTime(dateTime), internalAttr(0), externalAttr(0), uncompressedSize(0)
+QuaZipNewInfo::QuaZipNewInfo(const QString& _name, const QString& file, const QDateTime& _dateTime):
+  name(_name), dateTime(_dateTime), internalAttr(0), externalAttr(0), uncompressedSize(0)
 {
   QFileInfo info(file);
   if (info.exists()) {

--- a/quazip/unzip.c
+++ b/quazip/unzip.c
@@ -1054,7 +1054,7 @@ local int unz64local_GetCurrentFileInfoInternal (unzFile file,
             /* ZIP64 extra fields */
             if (headerId == 0x0001)
             {
-                uLong uL;
+                uLong _uL;
 
                 if(file_info.uncompressed_size == (ZPOS64_T)0xFFFFFFFFu)
                 {
@@ -1078,7 +1078,7 @@ local int unz64local_GetCurrentFileInfoInternal (unzFile file,
                 if(file_info.disk_num_start == 0xFFFFFFFFu)
                 {
                     /* Disk Start Number */
-                    if (unz64local_getLong(&s->z_filefunc, s->filestream,&uL) != UNZ_OK)
+                    if (unz64local_getLong(&s->z_filefunc, s->filestream, &_uL) != UNZ_OK)
                         err=UNZ_ERRNO;
                 }
 
@@ -2151,3 +2151,4 @@ int ZEXPORT unzClearFlags(unzFile file, unsigned flags)
     s->flags &= ~flags;
     return UNZ_OK;
 }
+

--- a/quazip/zip.c
+++ b/quazip/zip.c
@@ -645,6 +645,14 @@ local ZPOS64_T zip64local_SearchCentralDir64(const zlib_filefunc64_32_def* pzlib
   return relativeOffset;
 }
 
+// compilers hate this ONE SIMPLE TRICK!
+static int LoadCentralDirectoryRecord(zip64_internal* pziinit);
+static int Write_LocalFileHeader(zip64_internal* zi, const char* filename, uInt size_extrafield_local, const void* extrafield_local, uLong version_to_extract);
+static int Write_Zip64EndOfCentralDirectoryLocator(zip64_internal* zi, ZPOS64_T zip64eocd_pos_inzip);
+static int Write_Zip64EndOfCentralDirectoryRecord(zip64_internal* zi, uLong size_centraldir, ZPOS64_T centraldir_pos_inzip);
+static int Write_EndOfCentralDirectoryRecord(zip64_internal* zi, uLong size_centraldir, ZPOS64_T centraldir_pos_inzip);
+static int Write_GlobalComment(zip64_internal* zi, const char* global_comment);
+
 int LoadCentralDirectoryRecord(zip64_internal* pziinit)
 {
   int err=ZIP_OK;

